src/main/java/de/tum/in/www1/artemis/web/rest/lecture/LectureUnitResource.java

```diff
@@ -87,6 +93,40 @@ public ResponseEntity<List<LectureUnit>> updateLectureUnitsOrder(@PathVariable L
         return ResponseEntity.ok(persistedLecture.getLectureUnits());
     }
 
+    /**
+     * POST lectures/:lectureId/lecture-units/:lectureUnitId/complete
+     *
+     * @param lectureId     the id of the lecture to which the unit belongs
+     * @param lectureUnitId the id of the lecture unit to mark as completed for the logged-in user
+     * @return the ResponseEntity with status 200 (OK)
+     */
+    @PostMapping("/lectures/{lectureId}/lecture-units/{lectureUnitId}/complete")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Void> completeLectureUnit(@PathVariable Long lectureUnitId, @PathVariable Long lectureId) {
+        log.info("REST request to mark lecture unit as completed: {}", lectureUnitId);
+        LectureUnit lectureUnit = lectureUnitRepository.findById(lectureUnitId).orElseThrow(() -> new EntityNotFoundException("lectureUnit"));
+
+        if (lectureUnit.getLecture() == null || lectureUnit.getLecture().getCourse() == null) {
+            throw new ConflictException("Lecture unit must be associated to a lecture of a course", "LectureUnit", "lectureOrCourseMissing");
+        }
+
+        if (!lectureUnit.getLecture().getId().equals(lectureId)) {
+            throw new ConflictException("Requested lecture unit is not part of the specified lecture", "LectureUnit", "lectureIdMismatch");
+        }
+
+        if (!lectureUnit.isVisibleToStudents()) {
+            throw new ConflictException("Requested lecture unit is not yet visible for students", "LectureUnit", "lectureUnitNotReleased");
+        }
```

Comment:
Sometimes an instructor also participates in an exercise before the release date. Could this also happen here with the lecture unit?